### PR TITLE
WIP: HACDOCS-453 adding security steps

### PR DESCRIPTION
**Pull request by missmesss**
HACDOCS-453: Docs need to include the ways how users can improve the security of their customized pipelines.

The 3 recommendations can be added to the "Customization" section or be published as a separate "Security" piece, up to @Chr1st1anSears.

@jc-berger When you have time, will you please have a look? The very last paragraph includes 2 passive voice instances,  they can be reworded easily (by repeating the phrasing from a sentence before that) but I thought 2 passive voices weren't adding ambiguity there exactly because the previous sentence just explained things. WDYT? Thanks in advance for a review!

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_upgrade_build_pipeline.adoc
Comment (jc-berger) [inline docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_upgrade_build_pipeline.adoc:52]: I think the links in the middle of the procedure will encourage users to leave the procedure before completing it. I'll leave a comment at the end of my review, including the links in an "Additional resources" section, and you can see if you like that flow better. 

So, I'd recommend this step simply being:

1. Add an `OWNERS` file to the `.tekton` directory and list trusted contributors there.

**Review by jc-berger — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_upgrade_build_pipeline.adoc
Comment (missmesss) [reply docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_upgrade_build_pipeline.adoc:52]: I don't mind doing it either way. My only concern is that "Additional resources" at the end will refer to the whole guide (Upgrading the build pipeline) and not to this Security section only.

This new content was meant to become a part of [Upgrading the build pipeline](https://redhat-appstudio.github.io/docs.stonesoup.io/Documentation/main/how-to-guides/proc_upgrade_build_pipeline/) Christian mentioned he'd restructure this guide this week so probably he'll find a nice way of handling this content and the links?

File: docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_upgrade_build_pipeline.adoc
Comment (missmesss) [reply docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_upgrade_build_pipeline.adoc:49]: I only tried to match other noun-based titles within the guide like .Customization, but if this new content becomes a separate doc somewhere, I'll be happy to rename it as I agree it looks too minimalistic now :)

**Comment by missmesss** [comment]
@jc-berger Thanks for a quick and thorough review, Jake. I've implemented some changes but haven't restructured the content since Christian has big plans for this guide. Hopefully after his big updates it'll be clearer how to treat this piece. Making it a separate module makes sense too.

Please, let me know if you feel the text is looking good for now, and then we can return to it after Christian's updates?

**Comment by missmesss** [comment]
@amfred Will you please review the text about custom pipelines vulnerability? [Our review doc with your comments](https://docs.google.com/document/d/1PVbAGk3NJDrByjh6A6odHm-ylRJ3JR9SeG30w5to02k) for your convenience.

This content is being added to the end of [Upgrading the build pipeline](https://redhat-appstudio.github.io/docs.stonesoup.io/Documentation/main/how-to-guides/proc_upgrade_build_pipeline/) for now. 
Please let me know if you prefer gdocs for text reviews as asciidoc syntax can make reading painful.
Thanks!

**Comment by missmesss** [comment]
@amfred Also, is there a chance that you know a QE who could review this? Our processes require reviews from both an SME and a QE. I'll ask in Slack ofc, just thought you maybe have a name already, thanks

File: docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc
Comment (amfred) [inline docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:64]:
```suggestion
If you don’t set `pipelinerun_provenance`, you allow the default behaviour: the PipelineRun definition is fetched from the branch where the PipelineRun event is triggered. Testing changes to the PipelineRun is easier with the default behaviour, because PipelineRun changes will be tested when a Pull Request is submitted, before merging the change.  Setting the `pipelinerun_provenance` setting to `default_branch` is safer, because a PipelineRun definition will not be tested until it has been reviewed and merged by a repository owner. 
```

File: docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc
Comment (amfred) [inline docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:62]:
```suggestion
* Consider changing the PipelineRun definition source by setting the `pipelinerun_provenance` setting to `default_branch`. With this setting, Pipelines as Code uses the PipelineRun definition from the default branch of the repository (usually `main` or `master`), and only contributors with default branch merge rights can modify the PipelineRun.
```

**Review by amfred — CHANGES_REQUESTED** [review]

**Comment by missmesss** [comment]
@amfred Thank you a lot for a quick and thorough review. I mostly kept your wording, only eliminated a couple of passive voices and added some white space. WDYT?

Here's [a gdoc with the same text](https://docs.google.com/document/d/13o3cJp0s0nVLPYi7KSyKiicmjKc5S2nr6uGSAoYG1W0), needed it for my team, maybe you'll prefer it too.

File: docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc
Comment (amfred) [reply docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:58]: Really?  We have OWNERS files at various levels in infra-deployments.

Comment (chmouel) [reply docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:58]: that's for prow not for paac.

**Comment by missmesss** [comment]
Thanks for a review @chmouel , I've added info about root.

File: docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc
Comment (Chr1st1anSears) [inline docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:56]: Could we add "further" here? "To further reinforce the security of your custom build pipeline..." Because the earlier part of the doc states that you customize in the first place for stronger security. So I think we should clarify that this section tells readers how to reinforce their security even more :)

**Review by jc-berger — APPROVED** [review]
/lgtm

**Review by amfred — APPROVED** [review]
Looks good, and a few people from the Pipelines teams also reviewed it offline.

**Comment by missmesss** [comment]
> Looks good, and a few people from the Pipelines teams also reviewed it offline.

Thanks for lgtm and extra eyes on this :)

File: docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc
Comment (chmouel) [inline docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:75]: it doesn't prevent as well the fact that if a negligent owner merge a malicious changes to the default branch, it will be as unsecure...... 

maybe we want to add that the owner will still to be very careful before pressing the merge button by carefully reviewing the change to the pipelineruns.

File: docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc
Comment (chmouel) [inline docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:71]:
```suggestion
If you don’t set `pipelinerun_provenance`, you allow the default behaviour: the PipelineRun definition is fetched from the branch where the PipelineRun event is triggered, and any contributors can get access to the PipelineRun definition.
```

you may want to define what is the meaning of a contributor in pac terms, it means someone who is a contributor on a repository and any members of the organisation where the repository belongs.

Comment (missmesss) [reply docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:71]: @chmouel Thanks for being attentive to details, appreciate it. 
There's also a longer list of who can run pipelinerun on CI: https://pipelinesascode.com/docs/guide/running/ (in the beginning). In our case does this whole list apply? If yes, do you think we can just link to this list? To avoid rewriting a long piece of text that already exists in other docs.

File: docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc
Comment (missmesss) [reply docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:75]: OK, I'll add a sentence about extra careful reviewing, thanks.

File: docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc
Comment (chmouel) [reply docs/modules/ROOT/pages/how-to-guides/proc_upgrade_build_pipeline.adoc:71]: @missmesss yes, all those rules would applies on appstudios you can link there!

**Comment by missmesss** [comment]
@chmouel I'll appreciate another look. If it helps, there's a gdoc where new pieces are highlighted https://docs.google.com/document/d/13o3cJp0s0nVLPYi7KSyKiicmjKc5S2nr6uGSAoYG1W0. Thank you.

**Review by chmouel — APPROVED** [review]
/lgtm
/approve

**Review by jc-berger — APPROVED** [review]
/lgtm

**Pull request merged**
